src/Microsoft.TemplateEngine.Abstractions/IParameterSymbolLocalizationModel.cs

```diff
@@ -16,8 +16,14 @@ public interface IParameterSymbolLocalizationModel
         /// </summary>
         string? DisplayName { get; }
 
+        /// <summary>
+        /// Gets the localized description of the symbol.
+        /// </summary>
         string? Description { get; }
 
+        /// <summary>
+        /// Gets the localization models for choices of this symbol.
+        /// </summary>
         IReadOnlyDictionary<string, ParameterChoiceLocalizationModel> Choices { get; }
```

Comment:
Having this null would save a bit on allocations, but I think it makes consumption of this property easier since you don't have to check for nulls. It is also easier for deserializer to just create the dictionary in advance, instead of creating it if needed here: 
https://github.com/dotnet/templating/blob/bb688b93882c20f34ae00493bf2be4711bc1e221/src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/LocalizationModelDeserializer.cs#L108-L124

So I'd say we keep it non-nullable. Let me know what you think.